xbmc/utils/StringUtils.h

```diff
@@ -46,6 +46,9 @@
 #include "XBDateTime.h"
 #include "utils/params_check_macros.h"
 
+#define STR_HELPER(x) #x
+#define STR(x) STR_HELPER(x)
```

Comment:
what's this for?

Reply:
for converting a compiler definition to string, see gbm.cmake

Reply:
I understand that, but why don't you add the defines in xbmc/settings/Settings.cpp?

Reply:
IMO these 2 macros are StringUtils, but I don't insist on that location.
Any place is fine for me.

Reply:
Should heavily comment this usage of pre-processor magic. And it should have a very descriptive name that reflects the specialized use case. "STR" is way too generic.

Reply:
If you expect that this marco will be needed at other places too, keep it here, but it definitely has to be commented.